File: data/python/tests/test_lightgbm_feature_importance.py
import unittest
from function_lightgbm_feature_importance import FeatureImportance

class TestLightGBMFeatureImportance(unittest.TestCase):
    def setUp(self):
        self.feature_importance = FeatureImportance()

    def test_feature_importance_length(self):
        input_str = "100,5"
        result = self.feature_importance.quick_sort_from_string(input_str)
        importance_scores = list(map(int, result.split(',')))
        self.assertEqual(len(importance_scores), 5)

    def test_feature_importance_non_negative(self):
        input_str = "100,5"
        result = self.feature_importance.quick_sort_from_string(input_str)
        importance_scores = list(map(int, result.split(',')))
        self.assertTrue(all(score >= 0 for score in importance_scores))

    def test_feature_importance_sum(self):
        input_str = "100,5"
        result = self.feature_importance.quick_sort_from_string(input_str)
        importance_scores = list(map(int, result.split(',')))
        self.assertGreater(sum(importance_scores), 0)

    def test_feature_importance_type(self):
        input_str = "100,5"
        result = self.feature_importance.quick_sort_from_string(input_str)
        self.assertIsInstance(result, str)

    def test_feature_importance_large_number_of_features(self):
        input_str = "100,1000"
        result = self.feature_importance.quick_sort_from_string(input_str)
        importance_scores = list(map(int, result.split(',')))
        self.assertEqual(len(importance_scores), 1000)

if __name__ == '__main__':
    unittest.main()

File: data/java/tests/test_lightgbm_feature_importance.java
package com.example;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;
import java.util.Arrays;

public class test_lightgbm_feature_importance {
    private function_lightgbm_feature_importance feature_importance;

    @BeforeEach
    void setUp() {
        feature_importance = new function_lightgbm_feature_importance();
    }

    @Test
    void test_feature_importance_length() {
        String input_str = "100,5";
        String result = feature_importance.quick_sort_from_string(input_str);
        String[] importance_scores = result.split(",");
        assertEquals(5, importance_scores.length);
    }

    @Test
    void test_feature_importance_non_negative() {
        String input_str = "100,5";
        String result = feature_importance.quick_sort_from_string(input_str);
        boolean allNonNegative = Arrays.stream(result.split(","))
                .mapToInt(Integer::parseInt)
                .allMatch(score -> score >= 0);
        assertTrue(allNonNegative);
    }

    @Test
    void test_feature_importance_sum() {
        String input_str = "100,5";
        String result = feature_importance.quick_sort_from_string(input_str);
        int sum = Arrays.stream(result.split(","))
                .mapToInt(Integer::parseInt)
                .sum();
        assertTrue(sum > 0);
    }

    @Test
    void test_feature_importance_type() {
        String input_str = "100,5";
        String result = feature_importance.quick_sort_from_string(input_str);
        assertInstanceOf(String.class, result);
    }

    @Test
    void test_feature_importance_large_number_of_features() {
        String input_str = "100,1000";
        String result = feature_importance.quick_sort_from_string(input_str);
        String[] importance_scores = result.split(",");
        assertEquals(1000, importance_scores.length);
    }
}